docs/explainer.md

```diff
@@ -388,27 +391,159 @@ let message = decoder.decode(data);
 
 See [`WritableStream`](https://streams.spec.whatwg.org/#ws-intro) spec for more examples.
 
+##### Connected mode
+
 ```javascript
-let udpSocket = new UDPSocket(...);
+let udpSocket = new UDPSocket({ remoteAddress, remotePort });
 let { writable } = await udpSocket.opened;
 
 let writer = writable.getWriter();
 
 const encoder = new TextEncoder();
 let message = "Some user-created datagram";
 
-// sends a UDPMessage object with data = Uint8Array
+// Sends a UDPMessage object where |data| is a Uint8Array.
+// Note that |remoteAddress| and |remotePort| must not be specified.
 await writer.ready;
 writer.write({
     data: encoder.encode(message)
 }).catch(err => console.log(err));
 
-// or, alternatively, with data = ArrayBuffer
+// Sends a UDPMessage object where |data| is an ArrayBuffer.
+// Note that |remoteAddress| and |remotePort| must not be specified.
 await writer.ready;
 writer.write({
     data: encoder.encode(message).buffer
 }).catch(err => console.log(err));
 ...
 
 // Don't forget to call releaseLock() or cancel()/abort() on the writer once done.
-```
\ No newline at end of file
+```
+
+##### Bound mode
+
+```javascript
+let udpSocket = new UDPSocket({ localAddress });
+let { writable } = await udpSocket.opened;
+
+let writer = writable.getWriter();
+
+const encoder = new TextEncoder();
+let message = "Some user-created datagram";
+
+// Sends a UDPMessage object where |data| is a Uint8Array.
+// Note that both |remoteAddress| and |remotePort| must be specified in this case.
+await writer.ready;
+writer.write({
+    data: encoder.encode(message),
+    remoteAddress: 'example.com',
```

Comment:
Add a note that specifying a domain name as the `remoteAddress` for an unconnected socket is inefficient as it requires a DNS lookup for each packet. Applications should try to replace the domain with the IP address of the peer after receiving a response packet.